### ISSUE TITLE
jsDoc undefined type does not work

### ISSUE THREAD

**Issue by anacierdem** [microsoft/TypeScript#50926]
# Bug Report

### 🔎 Search Terms

jsdoc, undefined

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about jsDoc & js

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?noUnusedLocals=true&noUnusedParameters=true&declaration=false&target=1&module=1&exactOptionalPropertyTypes=true&ts=4.9.0-dev.20220921&filetype=js#code/ATD0Cp2ABAXBPADgU2AbwM6wE4EsB2A5sAD7ACu+AJsgGYHJUC+w4oAsAFAgDGA9vizAasZD1FUAyjgCGowvGABeCtToMqAbi5cQIsROnY5yBQDoeACxnYAgrAAUABgCUmoA)

### 💻 Code

```js
  /** @type {string | undefined} */
  const detectedStrategy = undefined;
  detectedStrategy.charAt(0);
```

### 🙁 Actual behavior

`detectedStrategy` is considered a string after defining its type in a js context.

### 🙂 Expected behavior

It should create a TS error when trying to access the string method on it. The error is generated is type is defined as `/** @type {string | void} */` but this does not work for implicit function return values as they are undefined by default.

**Comment by MartinJohns** [comment]
You have `strictNullChecks` disabled. Without this flag there is no `undefined` or `null` type.

**Comment by anacierdem** [comment]
You are right, thank you!
